app/src/main/java/pl/droidevs/books/library/LibraryActivity.java

```diff
@@ -292,6 +299,39 @@ private void requestWriteStoragePermissions() {
                 REQUEST_PERMISSION_SAVE_FILE_CODE);
     }
 
+    void importOptionSelected() {
+
+        if (hasReadStoragePermission()) {
+            importLibrary();
+        } else {
+            requestReadStoragePermissions();
+        }
+    }
+
+    private boolean hasReadStoragePermission() {
+        return ContextCompat.checkSelfPermission(this, Manifest.permission.READ_EXTERNAL_STORAGE)
+                == PackageManager.PERMISSION_GRANTED;
+    }
+
+    private void importLibrary() {
+        final ExportImportViewModel exportImportViewModel = ViewModelProviders
+                .of(this, viewModelFactory)
+                .get(ExportImportViewModel.class);
+
+        try {
+            exportImportViewModel.importBooks();
```

Comment:
Czy ta operacja blokuje UI? A jeśli nie to wiadomość pojawi się zanim import faktycznie się skończy. Warto by również pokazać progress bar na czas jej trwania.